src/bar/BarChartWithErrors.ts

```diff
@@ -0,0 +1,64 @@
+import {Bar} from 'vue-chartjs'
+import ErrorBarsPlugin from 'chartjs-plugin-error-bars'
+import Component, {mixins} from "vue-class-component";
+import {Prop, Watch} from "vue-property-decorator";
+import {BarChartData} from "./utils";
+
+@Component
+export default class BarChartWithErrors extends mixins(Bar) {
```

Comment:
Experimenting with the class API - ended up doing this because there's no way to make type checking work with:

```
Vue.extend({
   extends: "Bar"
})
```
What do you reckon to this syntax in general? Seems like quite a lot of people using Typescript take this approach Of course Vue3 is coming soon with another (functional) API in the mix as well! :grimacing:

Reply:
Ha, yeah it's a bit weird, but if it does the job... Maybe include a brief explanatory comment?

Reply:
You think it's worse than the `Vue.extend({})` syntax? I'm not sure, I think I might prefer it :thinking:

Reply:
Might just be me misunderstanding mixins - I thought mixins were like reusable fragments of components but tended not to be components in their own right as here.

Reply:
The mixins bit is weird I agree, but there's no equivalent function "extends(Bar)". And then there's the 2 uses of "extend" - `Vue.extend` vs `{extends: Bar}`. Overall Vue mixins/extensions just don't play that nicely with Typescript at all!

But mixins aside, I'm curious to know what you think of this class syntax as opposed to the object syntax? I can see pros and cons, but I think it might be a bit clearer.

Reply:
I like it - it does seem clearer, and putting the semantics of something into a decorator rather than the nesting of an object just feels like you get nicer looking code. I think we're pretty used to the object notation now so it's not really getting in the way, but happy to start using this approach instead !